src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+        self.no_preprune = no_preprune
+        self.leaf_prediction = leaf_prediction
+        self.nb_threshold = nb_threshold
+        self.nominal_attributes = nominal_attributes
+        self.learning_ratio_perceptron = learning_ratio_perceptron
+        self.learning_ratio_decay = learning_ratio_decay
+        self.learning_ratio_const = learning_ratio_const
+        self.random_state = random_state
+
+    def partial_fit(self, X, y, sample_weight=None):
+        """ Partially (incrementally) fit the model.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The features to train the model.
+
+        y: numpy.ndarray of shape (n_samples)
+            An array-like with the class labels of all samples in X.
+
+        sample_weight: numpy.ndarray of shape (n_samples), optional
+            (default=None)
+            Samples weight. If not provided, uniform weights are assumed.
+
+        Returns
+        -------
+        self
+
+        """
+        if sample_weight is None:
+            weight = 1.0
+        else:
+            weight = sample_weight
+
+        if y is not None:
+            row_cnt, _ = get_dimensions(X)
+            weight = check_weights(weight, expand_length=row_cnt)
+            for i in range(row_cnt):
+                if weight[i] != 0.0:
+                    self._train_weight_seen_by_model += weight[i]
+                    self._partial_fit(X[i], y[i], weight[i])
+
+        return self
+
+    def _partial_fit(self, X, y, sample_weight=1.0):
+        self.instances_seen += 1
+        
+        if self.ensemble is None:
+            self.init_ensemble(X)
+
+        for i in range(self.n_estimators):
+            y_predicted = self.ensemble[i].predict(np.asarray([X]))
+            self.ensemble[i].evaluator.add_result(y_predicted, y)
+            k = self._random_state.poisson(self.lambda_value)
+            if k > 0:
+                self.ensemble[i].partial_fit(np.asarray([X]), np.asarray([y]),
+                                             sample_weight=np.asarray([k]),
+                                             instances_seen=self.instances_seen)
+    
+    def predict(self, X):
+        """ Predict target value for the passed data.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The set of data samples for which to predict the target value.
+
+        Returns
+        -------
+        A numpy.ndarray with all the predictions for the samples in X.
+
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
+                # Ensemble is empty, all classes equal, default to zero
+                predictions.append(0)
+            else:
+                predictions.append(max(votes, key=votes.get))
+        return np.asarray(predictions)
```

Comment:
I believe that in the regression case an average aggregation function should be used.

Reply:
Indeed, I don't know how I could have overlooked this! It now correctly uses average aggregation instead.
@jacobmontiel , to do this I replaced get_votes_for_instance by _predict, in order to use the barycenter in the case of weighted_vote. Is that ok?

Reply:
The change is available in later commits:  https://github.com/scikit-multiflow/scikit-multiflow/blob/c73c759b559a76f66dfc95c9997336af712b4adb/src/skmultiflow/meta/adaptive_random_forest_regressor.py#L275-L292

Using `_predict` seems a good approach, although I have some comments:
- Renaming `votes` to something like `predicted_value` would make things even clearer.
- The MOA implementation does not take into account the performance to `weight` the predicted values. Comments @hmgomes?

Reply:
* Indeed, & that's now done.
* It just felt the natural way of weighting the predictions, but if it's not required we can remove the `disable_weighted_votes` parameter altogether, then.